zipline/assets/_assets.pyx

```diff
@@ -298,6 +313,80 @@ cdef class Future(Asset):
         super_dict['contract_multiplier'] = self.contract_multiplier
         return super_dict
 
+"""
+TODO MB - rework this we need a Currency object and we need a FX pair
+
+ok we should use this a CurrencyPair, Currency could just be an enum?
```

Comment:
Yes I was trying to think of a way to encapsulate this behaviour as we are dealing with a translation of currencies and not an FX pair you would trade so to speak. Having a base_currency on all assets would be ideal then really all you need is some kind of translation map between them. for some pairs that don't match 1:1 you may need to get creative to get a mapping between them but wouldn't be hard in a recursive loop around the model. Probably no more than one or two steps to get from TRY to SGD perhaps. (TRY->USD->SGD)